doc/frequency_evaluation_framework.md

```diff
@@ -0,0 +1,373 @@
+
+To support stratification, a new sketch type will be implemented,
+`StratifiedSketch`. A `StratifiedSketch` will be constructed from a list of
+cardinality sketch instances. The underlying method used for computing
+frequencies will depend upon the capabilities of the sketch instances that are
+passed to it. The `StratifiedSketch` will support providing frequency estimates
+for unions and intersections. In addition, the `StratifiedSketch` will have a
+method for instantiating a new instance from an instance of an `ExactSet`.
+
+#### Data Generator
+
+The
+[Cardinality and Frequency Estimation Evaluation Framework](https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/doc/cardinality_and_frequency_estimation_evaluation_framework.md)
+doc foresees three scenarios for the evaluation of frequency estimators:
+
+*   Scenario 1: Homogeneous user activities within a publisher
+*   Scenario 2: Heterogeneous user frequency
+*   Scenario 3: Publisher Constant Frequency
+
+Each of these will be implemented as subclasses of
+<code>[SetGeneratorBase](https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/src/simulations/set_generator.py)</code>.
+
+#### Configs
+
+A sample config will be created that demonstrates the working of the system with
+the
+<code>[ExactSet](https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/src/estimators/exact_set.py)</code>
+estimator.
+
+### Thoughts and Issues
+
+
+#### How will noising be handled?
+
+A single id will only impact a single bar in the frequency histogram, and
+deleting that id can change the value of the bar by at most one. Therefore, if
+the Laplace mechanism is used to add &epsilon;-differentially private noise to
+each bar of the histogram, the result will be &epsilon;-differentially private.
+
+The same considerations apply to the geometric mechanism. Since the existing
+implementations of the <code>LaplaceEstimateNoiser</code> and
+<code>GeometricEstimateNoiser</code> are able to add noise to a vector of
+values, no additional changes need to be made to support adding differential
+private noise to a frequency histogram.
+
+#### Can development proceed in parallel?
+
+After the
+<code>[EstimatorBase](https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/src/estimators/base.py)</code>
+is updated, it will be possible to implement the VoC and ADBC sketch types in
+parallel with the implementation of the remainder of the frequency evaluation
+framework. In addition, the
+<code>[SetGenerator](https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/src/simulations/set_generator.py)</code>'s
+for the different evaluation scenarios can be implemented in parallel.
+
+#### How will this support parallel computation?
```

Comment:
(cc. @zxhuang @my2457) Miao and Zeke, please take a look at this section and also for your reference.